census.go

```diff
@@ -468,6 +468,147 @@ func (v *vocdoniHandler) censusFollowers(msg *apirest.APIdata, ctx *httprouter.H
 	return ctx.Send(data, http.StatusOK)
 }
 
+// censusCommunity creates a new census from a community. The census of the
+// community can be of type channel, NFT, or ERC20. If the community is a
+// channel, the census is created from the users who follow the channel, and
+// the process is async. If the community is an NFT or ERC20, the census is
+// created from the token holders of the token addresses in the community, using
+// the AirStack API. The process is sync and the census is created in the same
+// request. The census is created from the participants and the progress is
+// updated in the queue.
+func (v *vocdoniHandler) censusCommunity(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	req := struct {
+		CommunityID uint64 `json:"communityID"`
+	}{}
+	if err := json.Unmarshal(msg.Data, &req); err != nil {
+		return err
+	}
+	// get the community from the database
+	community, err := v.db.Community(req.CommunityID)
+	if err != nil {
+		return ctx.Send([]byte("error getting community"), http.StatusInternalServerError)
+	}
+	if community == nil {
+		return ctx.Send([]byte("community not found"), http.StatusNotFound)
+	}
+	// create a censusID for the queue and store into it
+	censusID, err := v.cli.NewCensus(api.CensusTypeWeighted)
+	if err != nil {
+		return err
+	}
+	// check the type to create it from the correct source (channel or airstak
+	// (nft/erc20)) and in the correct way (async or sync)
+	if community.Census.Type == mongo.TypeCommunityCensusChannel {
```

Comment:
Please move this code to a separate function and try to reuse the one already existing for creating channel censuses.